src/lib.rs

```diff
@@ -601,6 +598,13 @@ async fn start_synapse_container(
                             typ: Some("json-file".to_string()),
                             config: None,
                         }),
+                        // Synapse has a tendency to not start correctly
+                        // or to stop shortly after startup. The following
+                        // restart policy seems to help a lot.
+                        restart_policy: Some(RestartPolicy {
+                            name: Some(RestartPolicyNameEnum::ALWAYS),
+                            maximum_retry_count: None,
```

Comment:
Do we know what can go wrong with no maximum retry counts.

Always seems like the right option though

Reply:
Good question.

This can fail if Synapse repeatedly can't start e.g. because of a problem in the module. Of course, atm, there's no way to check that. Perhaps we could detect this by adding a dedicated module to e.g. touch a given file when startup is complete. Maybe in a followup :)

For the time being, I'll probably add a retry count of 20 and hope that it works.